### DISCUSSION TITLE
How  to  trigger workflow run once a week if push has occurred?

### DISCUSSION THREAD

**Discussion by benstear** [community/community#26519] [Actions]
<p>I would like to trigger my github actions workflow to run once a week,only if there have been any pushes made to the repo in that week. Is there any way  to do this?</p>
<p>Thanks.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/benstear">@benstear</a> ,</p>
<p>If you want your workflow run weekly, you need to set <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule" rel="nofollow noopener"><strong>Scheduled events</strong></a> for the workflow.<br>
If you want the workflow runs only when have pushes in that week, you need to compare whether the <strong>commit SHA</strong> ( <strong>github.sha</strong> ) of current workflow run is different with that of the last workflow run.<br>
You can use <strong><a href="https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows" rel="nofollow noopener">cache action</a></strong> to store and pass the <strong>commit SHA</strong> in each workflow run.</p>
<p>A simple demo:</p>
<pre><code>on:
  schedule:
    - cron: '0 0 * * 0'

jobs:
  test_schedule:
    name: Test schedule
    runs-on: ubuntu-latest
    steps:
      - name: Create cache file
        run: |
          mkdir check-SHA
          echo ${{ github.sha }} &gt; github-sha.txt

      - name: Check SHA
        id: check_sha
        uses: actions/cache@v1
        with:
          path: check-SHA
          key: check-SHA-${{ github.sha }}

      - name: Some subsequent steps
        if: steps.check_sha.outputs.cache-hit != 'true'
        run: echo "Exccute the subsequent steps"
</code></pre>
<p>You can see I set the <strong>key</strong> of the cache contains the  <strong>commit SHA </strong> of current workflow run, when execute the <strong>cache action</strong> , it will check whether the specified <strong>key </strong> matchs an existing cache. If the <strong>key </strong> matchs an existing cache ( <strong>cache-hit</strong> is <strong>true</strong> ), it will restore this cache to the specified <strong>path</strong> on the local, and overwrite the file created in the " <strong>Create cache file</strong>" step in the <strong>path</strong>. If the <strong>key</strong> doesn’t match any existing cache ( <strong>cache-hit</strong> is <strong>false</strong> ), the action will save the file in the " <strong>Create cache file</strong>" step in the specified <strong>path </strong> as a cache with the specified <strong>key</strong>.</p>
<p>In each subsequent step, use the <strong>if conditional</strong> to skip the subsequent step when the  <strong>cache-hit</strong> is <strong>true</strong>.</p>

**Reply by EricZinda** [reply]
Is this line purely for debugging?
~~~
echo ${{ github.sha }} > github-sha.txt
~~~

**Comment by austenstone** [comment]
I made an action for that [austenstone/schedule](https://github.com/austenstone/schedule)
